### ISSUE TITLE
[release-1.20] Restoring from snapshot fails executable file not found in $PATH

### ISSUE THREAD

**Issue by ShylajaDevadiga** [rancher/rke2#1052]
**Environmental Info:**
RKE2 Version: 
* `v1.20.7+rke2r1`

Node(s) CPU architecture, OS, and Version: 
Ubuntu 20.04

Cluster Configuration:
Single node cluster

**Describe the bug:**
Restoring from snapshot fails with Containerd Exited: exec: "containerd": executable file not found in $PATH, exiting kubelet 

**Steps To Reproduce:**
Create a single node cluster
Take snapshot
Stop rke2 service
Run killall 
Restore using the snapshot
```
rke2 server \
--cluster-reset \
--cluster-reset-restore-path=/var/lib/rancher/rke2/server/db/snapshots/on-demand-<unix-timestamp
```

**Expected behavior:**
Reset should be run and cluster-reset msg is expected to be seen on the console

**Actual behavior:**
Errors with Containerd Exited: exec: "containerd": executable file not found in $PATH, exiting kubelet  and panic

**Additional context / logs:**
```
sudo rke2 server --cluster-reset --cluster-reset-restore-path=/var/lib/rancher/rke2/server/db/snapshots/on-demand-ip-172-31-39-67-1621896967
WARN[0000] not running in CIS mode                      
INFO[0000] starting kubelet to clean up old static pods 
INFO[0005] Containerd Exited: exec: "containerd": executable file not found in $PATH, exiting kubelet 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4be89b]

goroutine 1 [running]:
os.(*Process).signal(0x0, 0x70cc4b8, 0x9be7958, 0x0, 0x0)
	/usr/local/go/src/os/exec_unix.go:63 +0x3b
os.(*Process).Signal(...)
	/usr/local/go/src/os/exec.go:135
os.(*Process).kill(...)
	/usr/local/go/src/os/exec_posix.go:66
os.(*Process).Kill(...)
	/usr/local/go/src/os/exec.go:120
github.com/rancher/rke2/pkg/rke2.removeOldPodManifests(0x66d7963, 0x15, 0xc001699a70, 0x1, 0x0, 0x0)
	/source/pkg/rke2/rke2.go:229 +0x945
github.com/rancher/rke2/pkg/rke2.setup(0xc000a4b4a0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/source/pkg/rke2/rke2.go:164 +0x8a5
github.com/rancher/rke2/pkg/rke2.Server(0xc000a4b4a0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/source/pkg/rke2/rke2.go:48 +0x6b
github.com/rancher/rke2/pkg/cli/cmds.ServerRun(0xc000a4b4a0, 0x0, 0x0)
	/source/pkg/cli/cmds/server.go:143 +0x98
github.com/urfave/cli.HandleAction(0x57f1f00, 0x68fece0, 0xc000a4b4a0, 0xc000a4b4a0, 0x0)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:524 +0x105
github.com/urfave/cli.Command.Run(0x6693772, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0, 0x66da585, 0x15, 0xc0017b8a50, ...)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/command.go:173 +0x579
github.com/urfave/cli.(*App).Run(0xc0009afc00, 0xc000a52c00, 0x4, 0x4, 0x0, 0x0)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:277 +0x808
main.main()
	/source/main.go:20 +0x714
```

**Comment by brandond** [comment]
This can be reproduced on a clean install of RKE2 1.20.7+rke2r1. After starting RKE2 once with `rke2 server` to allow it to bootstrap the binaries, run `rke2 server --cluster-reset`:
```
[root@centos01 ~]# rke2 server --cluster-reset
WARN[0000] not running in CIS mode
INFO[0000] starting kubelet to clean up old static pods
INFO[0005] kubelet Exited: exec: "kubelet": executable file not found in $PATH, exiting Containerd
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4be89b]

goroutine 1 [running]:
os.(*Process).signal(0x0, 0x70cc4b8, 0x9be7958, 0x0, 0x0)
	/usr/local/go/src/os/exec_unix.go:63 +0x3b
os.(*Process).Signal(...)
	/usr/local/go/src/os/exec.go:135
os.(*Process).kill(...)
	/usr/local/go/src/os/exec_posix.go:66
os.(*Process).Kill(...)
	/usr/local/go/src/os/exec.go:120
github.com/rancher/rke2/pkg/rke2.removeOldPodManifests(0x66d7963, 0x15, 0xc000d1a6f0, 0x1, 0x0, 0x0)
	/source/pkg/rke2/rke2.go:225 +0xa6c
github.com/rancher/rke2/pkg/rke2.setup(0xc000d61b80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/source/pkg/rke2/rke2.go:164 +0x8a5
github.com/rancher/rke2/pkg/rke2.Server(0xc000d61b80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/source/pkg/rke2/rke2.go:48 +0x6b
github.com/rancher/rke2/pkg/cli/cmds.ServerRun(0xc000d61b80, 0x0, 0x0)
	/source/pkg/cli/cmds/server.go:143 +0x98
github.com/urfave/cli.HandleAction(0x57f1f00, 0x68fece0, 0xc000d61b80, 0xc000d61b80, 0x0)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:524 +0x105
github.com/urfave/cli.Command.Run(0x6693772, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0, 0x66da585, 0x15, 0xc00011fa28, ...)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/command.go:173 +0x579
github.com/urfave/cli.(*App).Run(0xc000d5c380, 0xc000798ea0, 0x3, 0x3, 0x0, 0x0)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:277 +0x808
main.main()
	/source/main.go:20 +0x714
```

Deleting `/var/lib/rancher/rke2/data/` and /var/lib/rancher/rke2/bin` before running `rke2 server --cluster-reset`, and/or running `rke2 server --debug --cluster-reset` appear to have odd effects as well, although neither give the desired result.

The root cause of the immediate error appears to be due to the change from #971 to stage the binaries after startup, not working well with the change from #955 to start up the kubelet in standalone mode in order to delete static pods. The agent bootstrap side-effect of changing the system PATH variable no longer occurs early enough for the standalone kubelet to function properly. It is unclear why this wasn't caught by normal release testing.

**Comment by davidnuzik** [comment]
@brandond @ShylajaDevadiga assigned to new v1.20.7+rke2r2 milestone. We will fix and release a new rke2 1.20.7 as soon as able.

**Comment by ShylajaDevadiga** [comment]
Validated on three server node cluster using v1.20.7-rc1+rke2r2
```
$ kubectl get nodes
NAME                                         STATUS   ROLES                       AGE   VERSION
ip-172-31-2-187.us-east-2.compute.internal   Ready    control-plane,etcd,master   85m   v1.20.7-rc1+rke2r2
ip-172-31-3-226.us-east-2.compute.internal   Ready    control-plane,etcd,master   84m   v1.20.7-rc1+rke2r2
ip-172-31-6-17.us-east-2.compute.internal    Ready    <none>                      84m   v1.20.7-rc1+rke2r2
ip-172-31-7-82.us-east-2.compute.internal    Ready    control-plane,etcd,master   88m   v1.20.7-rc1+rke2r2
```
```
sudo ETCDCTL_API=3 etcdctl --cert /var/lib/rancher/rke2/server/tls/etcd/server-client.crt --key /var/lib/rancher/rke2/server/tls/etcd/server-client.key --endpoints https://127.0.0.1:2379 --cacert /var/lib/rancher/rke2/server/tls/etcd/server-ca.crt member list
79d11213dfcb5c90, started, ip-172-31-7-82-90a93bb6, https://172.31.7.82:2380, https://172.31.7.82:2379
7a03249b631d8688, started, ip-172-31-2-187-7c194cf5, https://172.31.2.187:2380, https://172.31.2.187:2379
b9aa84479811de4e, started, ip-172-31-3-226-dd12d5f9, https://172.31.3.226:2380, https://172.31.3.226:2379
```
```
sudo rke2 etcd-snapshot
...
{"level":"info","msg":"saved","path":"/var/lib/rancher/rke2/server/db/snapshots/on-demand-ip-172-31-7-82-1622002087"}
INFO[0000] Saving current etcd snapshot set to rke2-etcd-snapshots ConfigMap 
```
Power off two nodes

Stop rke2 service on node 1
```
sudo systemctl stop rke2-server
```

Restore node1 using the snapshot
```
 sudo rke2 server --cluster-reset --cluster-reset-restore-path=/var/lib/rancher/rke2/server/db/snapshots/on-demand-ip-172-31-7-82-1622002087
...
INFO[0023] Updating load balancer server addresses -> [172.31.2.187:9345 127.0.0.1:9345] 
INFO[0023] Etcd is running, restart without --cluster-reset flag now. Backup and delete ${datadir}/server/db on each peer etcd server and rejoin the nodes 

```
$ sudo systemctl start rke2-server
```
$ kubectl get nodes
NAME                                         STATUS     ROLES                       AGE    VERSION
ip-172-31-2-187.us-east-2.compute.internal   NotReady   control-plane,etcd,master   101m   v1.20.7-rc1+rke2r2
ip-172-31-3-226.us-east-2.compute.internal   NotReady   control-plane,etcd,master   100m   v1.20.7-rc1+rke2r2
ip-172-31-6-17.us-east-2.compute.internal    NotReady   <none>                      100m   v1.20.7-rc1+rke2r2
ip-172-31-7-82.us-east-2.compute.internal    Ready      control-plane,etcd,master   104m   v1.20.7-rc1+rke2r2
```
Delete the two nodes in NotReady State
Join new nodes or uninstall and reinstall to join back the two nodes
